subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DependencyManagementBuildTreeScopeServices.java

```diff
@@ -200,6 +209,25 @@ ModuleRepositoryCacheProvider createModuleRepositoryCacheProvider(
         return new ModuleRepositoryCacheProvider(persistentCaches, inMemoryOnlyCaches);
     }
 
+    ImmutableTransformWorkspaceServices createImmutableTransformWorkspaceServices(
```

Comment:
❓ How does it work to have the immutable workspace build tree scoped while still allowing build logic dependencies to be transformed? Is it currently possible to run artifact transforms before the build tree scope services are ready?

Reply:
I should check that, I don't know the answer right now 🤔